smac/intensification/intensification.py

```diff
@@ -307,7 +163,242 @@ def intensify(self, challengers, incumbent, run_history, aggregate_func,
+                status, cost, dur, res = self.tae_runner.start(
+                    config=challenger,
+                    instance=instance,
+                    seed=seed,
+                    cutoff=cutoff,
+                    instance_specific=self.instance_specifics.get(instance, "0"))
+                self._num_run += 1
+
+                if status == StatusType.ABORT:
+                    self.logger.debug(
+                        "TAE signaled ABORT -- stop intensification on challenger")
+                    return incumbent
+
+            new_incumbent = self._compare_configs(
+                incumbent=incumbent, challenger=challenger, run_history=run_history,
+                aggregate_func=aggregate_func)
+            if new_incumbent == incumbent:
+                break
+            elif new_incumbent == challenger:
+                incumbent = challenger
+                break
+            else:  # Line 17
+                # challenger is not worse, continue
+                N = 2 * N
+
+        return incumbent
+
+    def _compare_configs(self, incumbent: Configuration, challenger: Configuration, run_history: RunHistory,
+                         aggregate_func: typing.Callable):
+        '''
+            compare two configuration wrt the runhistory 
+            and return the one which performs better (or None if the decision is not safe)
+
+            Decision strategy to return x as being better than y:
+                1. x has at least as many runs as y
+                2. x performs better than y on the intersection of runs on x and y
+
+            Implicit assumption: 
+                challenger was evaluated on the same instance-seed pairs as incumbent
+
+            Parameters
+            ----------
+            incumbent: Configuration
+                current incumbent
+            challenger: Configuration
+                challenger configuration
+            run_history: RunHistory
+                stores all runs we ran so far
+            aggregate_func: typing.Callable
+                aggregate performance across instances
+
+            Returns
+            -------
+            None or better of the two configurations x,y
+        '''
+
+        inc_runs = run_history.get_runs_for_config(incumbent)
+        chall_runs = run_history.get_runs_for_config(challenger)
+        
+        # performance on challenger runs
+        chal_perf = aggregate_func(
+            challenger, run_history, chall_runs)
+        inc_perf = aggregate_func(
+            incumbent, run_history, chall_runs)
+
+        # Line 15
+        if chal_perf > inc_perf and len(chall_runs) >= self.minR:
+            # Incumbent beats challenger
+            self.logger.debug("Incumbent (%.4f) is better than challenger (%.4f) on %d runs." % (
+                inc_perf, chal_perf, len(chall_runs)))
+            return incumbent
+
+        # Line 16
+        if len(chall_runs) == len(inc_runs):
```

Comment:
This isn't what is written in the docstring. The docstring states 'at least as many', thus it should be a `>=`.